### PR TITLE
Branch test

### PR DESCRIPTION
**Pull request by AnatSid**

### CODE REVIEW

File: src/main/java/AddCommand.java [hunk 1]
```diff
@@ -15,6 +15,13 @@ public String toString() {
 
     @Override
     public void execute() {
+        User user = this.createNewUser();
+        usersBook.addUser(user);
+        System.out.println("New user created:\n" + user + "\nUser ID: " + user.getId());
+        System.out.println(SEPARATOR);
+    }
+
+    public User createNewUser() {
```
Comment (KonstantinKizin) [inline src/main/java/AddCommand.java:24]: you don't need it.  You can extend a bit user book and add method to find last added user.
AddUserCommand.execute(); 
User user = userBook.getLastAddedUser()

File: src/main/java/EmailNotification.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import java.util.Scanner;
+
+public class EmailNotification implements Notifications {
+    @Override
+    public void sendNotification(String message) {
```
Comment (KonstantinKizin) [inline src/main/java/EmailNotification.java:5]: this class should only send message to target email. Method signature should be something like : sendNotification(String emailFrom, String emailTo, String message)

File: src/main/java/EmailUtil.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+
+public class EmailUtil {
```
Comment (KonstantinKizin) [inline src/main/java/EmailUtil.java:18]: No need, all this logic should be in your Notifications class

File: src/main/java/EmailUtil.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            msg.setSentDate(new Date());
+
+            msg.setRecipients(Message.RecipientType.TO, InternetAddress.parse(toEmail, false));
+
+            Transport.send(msg);
+
+            System.out.println("EMail Sent Successfully!!");
+
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (KonstantinKizin) [inline src/main/java/EmailUtil.java:53]: user won't be happy to see stack trace

File: src/main/java/Notifications.java [hunk 5]
```diff
@@ -0,0 +1,4 @@
+public interface Notifications {
```
Comment (KonstantinKizin) [inline src/main/java/Notifications.java:1]: usually such classes has Service as suffix. NotificationService in this case

File: src/main/java/TLSEmail.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
+
+import java.util.Properties;
+import javax.mail.Authenticator;
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+
+public class TLSEmail {
```
Comment (KonstantinKizin) [inline src/main/java/TLSEmail.java:7]: TLS what is it ?

File: src/main/java/AddAndNotifyCommand.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    @Override
+    public void execute() {
+        AddCommand add = new AddCommand(usersBook);
+        User user = add.createNewUser();
+        usersBook.addUser(user);
+
+        Notifications emailNotification = new EmailNotification();
+        emailNotification.sendNotification("You have successfully registered in our service. Your ID: "+ user.getId());
```
[the remark this answers is not in the document]
Comment (AnatSid) [reply src/main/java/AddAndNotifyCommand.java:17]: внутри метода sendNotification есть сканнер. который это спрашивает

**Pull request closed without merging**